boot/freeldr/freeldr/include/hwide.h

```diff
@@ -0,0 +1,321 @@
+#endif
+
+/*
+* IDE status definitions
+*/
+#define IDE_STATUS_SUCCESS           0x00
+#define IDE_STATUS_ERROR             0x01
+#define IDE_STATUS_INDEX             0x02
+#define IDE_STATUS_CORRECTED_ERROR   0x04
+#define IDE_STATUS_DRQ               0x08
+#define IDE_STATUS_DSC               0x10
+#define IDE_STATUS_DMA               0x20      /* DMA ready */
+#define IDE_STATUS_DWF               0x20      /* drive write fault */
+#define IDE_STATUS_DRDY              0x40
+#define IDE_STATUS_IDLE              0x50
+#define IDE_STATUS_BUSY              0x80
+
+#define IDE_STATUS_WRONG             0xff
+#define IDE_STATUS_MASK              0xff
+
+/*
+* IDE drive select/head definitions
+*/
+#define IDE_DRIVE_SELECT             0xA0
+#define IDE_DRIVE_1                  0x00
+#define IDE_DRIVE_2                  0x10
+#define IDE_DRIVE_SELECT_1           (IDE_DRIVE_SELECT | IDE_DRIVE_1)
+#define IDE_DRIVE_SELECT_2           (IDE_DRIVE_SELECT | IDE_DRIVE_2)
+#define IDE_DRIVE_MASK               (IDE_DRIVE_SELECT_1 | IDE_DRIVE_SELECT_2)
+#define IDE_USE_LBA                  0x40
+
+/*
+* IDE drive control definitions
+*/
+#define IDE_DC_DISABLE_INTERRUPTS    0x02
+#define IDE_DC_RESET_CONTROLLER      0x04
+#define IDE_DC_A_4BIT                0x80
+#define IDE_DC_USE_HOB               0x80 // use high-order byte(s)
+#define IDE_DC_REENABLE_CONTROLLER   0x00
+
+/*
+* IDE error definitions
+*/
+#define IDE_ERROR_ICRC               0x80
+#define IDE_ERROR_BAD_BLOCK          0x80
+#define IDE_ERROR_DATA_ERROR         0x40
+#define IDE_ERROR_MEDIA_CHANGE       0x20
+#define IDE_ERROR_ID_NOT_FOUND       0x10
+#define IDE_ERROR_MEDIA_CHANGE_REQ   0x08
+#define IDE_ERROR_COMMAND_ABORTED    0x04
+#define IDE_ERROR_END_OF_MEDIA       0x02
+#define IDE_ERROR_NO_MEDIA           0x02
+#define IDE_ERROR_ILLEGAL_LENGTH     0x01
+
+/*
+* Values for TransferMode
+*/
+#define ATA_PIO                      0x00
+
+/*
+* IDENTIFY data
+*/
+#include <pshpack1.h>
+typedef struct _IDENTIFY_DATA {
+    UCHAR  AtapiCmdSize:2;                  // 00 00 General configuration
+    UCHAR  :3;
+    UCHAR  DrqType:2;
+    UCHAR  Removable:1;
+    UCHAR  DeviceType:5;
+    UCHAR  :1;
+    UCHAR  CmdProtocol:2;
+    USHORT NumberOfCylinders;               // 02  1
+    USHORT Reserved1;                       // 04  2
+    USHORT NumberOfHeads;                   // 06  3
+    USHORT UnformattedBytesPerTrack;        // 08  4
+    USHORT UnformattedBytesPerSector;       // 0A  5
+    USHORT SectorsPerTrack;                 // 0C  6
+    USHORT VendorUnique1[3];                // 0E  7-9
+    USHORT SerialNumber[10];                // 14  10-19
+    USHORT BufferType;                      // 28  20
+    USHORT BufferSectorSize;                // 2A  21
+    USHORT NumberOfEccBytes;                // 2C  22
+    USHORT FirmwareRevision[4];             // 2E  23-26
+    USHORT ModelNumber[20];                 // 36  27-46
+    UCHAR  MaximumBlockTransfer;            // 5E  47
```

Comment:
> When the READ MULTIPLE command is issued, the Sector Count register contains the number of sectors
(not the number of blocks) requested.

Sector Count is 59, which is CurrentMultiSectorSetting in the current list.

47 is AKA Multiple Sector Command, Sector Count

Reply:
http://www.t13.org/Documents/UploadedDocuments/project/d1153r18-ATA-ATAPI-4.pdf, p. 79. I don't see a mistake.

Reply:
> Word 47: READ/WRITE MULTIPLE support

http://www.t13.org/documents/uploadeddocuments/docs2006/d1699r3f-ata8-acs.pdf

ACS-3 says

> Bits 7:0 of this word define the maximum number of logical sectors per DRQ data block that the device supports for READ MULTIPLE commands (see 7.26), READ MULTIPLE EXT commands (see 7.27), WRITE MULTIPLE commands (see 7.64), WRITE MULTIPLE EXT commands (see 7.65), and WRITE MULTIPLE EXT FUA commands (see 7.66).
For SATA devices, bits 7:0 shall be set to 16 or less.

Word 47 is obsolete since ACS-4.

tl;dr: it's known as **READ/WRITE MULTIPLE support**.

Reply:
Thank you, fixed.